packages/components/src/internal/components/samples/actions.ts

```diff
@@ -396,77 +394,85 @@ function getSamplesIdsNotFound(queryName: string, orderedIds: string[]): Promise
     });
 }
 
-export function getFindSamplesByIdData(): Promise<{ queryName: string; missingIds?: { [key: string]: string[] } }> {
+export function getFindSamplesByIdData(sessionKey: string): Promise<{ queryName: string; ids: string[], missingIds?: { [key: string]: string[] } }> {
     return new Promise((resolve, reject) => {
-        const ids = JSON.parse(sessionStorage.getItem(FIND_IDS_SESSION_STORAGE_KEY));
-        if (ids) {
+        Ajax.request({
+            url: ActionURL.buildURL('experiment', 'saveOrderedSamplesQuery.api'),
```

Comment:
I think you explained this during the call yesterday, but just to be sure. This `getFindSamplesByIdData` calls the `saveOrderedSamplesQuery`. But is it that since this case is just posting the sessionKey that this "save" API really is a get for that key?

Reply:
There are two parts of this getting saved in the session: the list of ids to search for (needed because of the ability to add more ids once you get to the "Find in Bulk" page) and the query created from those ids, which is what is being saved here.  The payload is does also contain the ids, so the name is a little weird maybe.  Suggestion?